jpa/multitenancy/partition/src/main/java/com/example/multitenancy/partition/config/tenant/TenantInterceptor.java

```diff
@@ -1,30 +1,44 @@
 package com.example.multitenancy.partition.config.tenant;
 
-import static org.springframework.http.HttpStatus.FORBIDDEN;
-
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.net.URI;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
 import org.springframework.lang.Nullable;
 import org.springframework.util.StringUtils;
 import org.springframework.web.servlet.HandlerInterceptor;
+import tools.jackson.databind.ObjectMapper;
 
 @Configuration(proxyBeanMethods = false)
 public class TenantInterceptor implements HandlerInterceptor {
 
     private final TenantIdentifierResolver tenantIdentifierResolver;
+    private final ObjectMapper objectMapper;
 
-    public TenantInterceptor(TenantIdentifierResolver tenantIdentifierResolver) {
+    public TenantInterceptor(TenantIdentifierResolver tenantIdentifierResolver, ObjectMapper objectMapper) {
         this.tenantIdentifierResolver = tenantIdentifierResolver;
+        this.objectMapper = objectMapper;
     }
 
     @Override
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
             throws IOException {
         var tenant = request.getParameter("tenant");
         if (request.getServletPath().startsWith("/api/") && !StringUtils.hasText(tenant)) {
-            response.sendError(FORBIDDEN.value(), "Unknown user tenant");
+            ProblemDetail problemDetail =
+                    ProblemDetail.forStatusAndDetail(HttpStatus.FORBIDDEN, "Unknown Database tenant");
```

Comment:
@coderabbitai, can you please help me to write junit for showcasing this scenario